pkgs/development/compilers/alan/default.nix

```diff
@@ -0,0 +1,57 @@
+{ stdenv, lib, fetchFromGitHub,
+  cgreen, openjdk, pkg-config, which
+}:
+
+let
+  pname = "alan";
+  version = "3.0beta8";
+
+in stdenv.mkDerivation {
+
+  name = "${pname}-${version}";
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    owner = "alan-if";
+    repo = "alan";
+    rev = "v${version}";
+    sha256 = "sha256:0zfg1frmb4yl39hk8h733bmlwk4rkikzfhvv7j34cxpdpsp7spzl";
+  };
+
+  postPatch = ''
+    # Some build-time scripts have inappropriate shebangs
+    patchShebangs --build bin
+    # The Makefiles have complex CFLAGS that don't allow separate control of optimization
+    sed -i 's/-O0/-O2/g' compiler/Makefile.common
+    sed -i 's/-Og/-O2/g' interpreter/Makefile.common
+  '';
+
+  installPhase = ''
+    bin=$out/bin
+    share=$out/share/alan
+    examples=$share/examples
+    mkdir -p $bin $share $examples
+    # Build the release tarball
+    make package
+    # The release tarball isn't split up into subdirectories
+    tar -xf alan*.tgz --strip-components=1 -C $share
+    mv $share/*.alan $examples
+    chmod a-x $examples/*.alan
+    mv $share/{alan,arun} $bin
+    # a2a3 isn't included in the release tarball
+    cp bin/a2a3 $bin
```

Comment:
```suggestion
    mkdir -p $out/bin $out/share/alan/examples
    # Build the release tarball
    make package
    # The release tarball isn't split up into subdirectories
    tar -xf alan*.tgz --strip-components=1 -C $out/share/alan
    mv $share/*.alan $out/share/alan/examples
    chmod a-x $out/share/alan/examples/*.alan
    mv $out/share/alan/{alan,arun} $out/bin
    # a2a3 isn't included in the release tarball
    cp bin/a2a3 $out/bin
```

Reply:
I used the extra variables here because the process of moving things around in the out directory is cumbersome and using the longer paths makes it hard to read.

Also, there's a mistake in one place in your suggestions, where `$share` is still used but it's no longer defined.

Reply:
> using the longer paths makes it hard to read.

Those paths are really short. No need to hide things behind additional variables.

Reply:
I've made these changes now.